autosubmit_api/experiment/common_requests.py

```diff
@@ -489,29 +493,33 @@ def get_experiment_log_last_lines(expid):
     error = False
     error_message = ""
     logcontent = []
-    reading = ""
 
     try:
         APIBasicConfig.read()
         exp_paths = ExperimentPaths(expid)
-        path = exp_paths.tmp_as_logs_dir
-        reading = os.popen('ls -t ' + path + ' | grep "run.log"').read() if (os.path.exists(path)) else ""
 
-        # Finding log files
-        if len(reading) == 0:
-            path = APIBasicConfig.LOCAL_ROOT_DIR + '/' + expid + '/' + APIBasicConfig.LOCAL_TMP_DIR
-            reading = os.popen('ls -t ' + path + ' | grep "run.log"').read() if (os.path.exists(path)) else ""
+        dir_files = []
+        path = None
+
+        # Try to read from the tmp_as_logs folder
+        if os.path.exists(exp_paths.tmp_as_logs_dir):
```

Comment:
Might be easier if we used `pathlib` instead of `os` and `os.path` where possible. e.g. (not tested)

```py
import pathlib

logs_dir = Path(exp_paths.tmp_as_logs_dir)
tmp_dir = Path(exp_paths.tmp_dir)

path: Optional[Path] = None
dir_files: list[Path] = []

if logs_dir.exists():
    dir_files = get_files_from_dir_with_pattern(tmp_dir, 'run.log')
    path = logs_dir

if not dir_files and tmp_dir.exists():
    dir_files = get_files_from_dir_with_pattern(tmp_dir, 'run.log')
    path = tmp_dir

if dir_files:
    log_file = Path(dir_files[0], log_file_name)
    timest = int(log_file.stat().st_mtime)
    ...
```

But that can be done later, if at all.

Reply:
Yes, I have in mind to move from `os.path` to `pathlib` later but for now we must try to minimize changes

Reply:
Sounds good to me!